src/ImageComparator.php

```diff
@@ -347,4 +352,62 @@ private function processImagePixels(
 
         return $pixels;
     }
+
+    public function calculateRGBDistance($color1, $color2): float
+    {
+        $r1 = $color1['r'];
+        $g1 = $color1['g'];
+        $b1 = $color1['b'];
+
+        $r2 = $color2['r'];
+        $g2 = $color2['g'];
+        $b2 = $color2['b'];
+
+        return sqrt(pow($r1 - $r2, 2) + pow($g1 - $g2, 2) + pow($b1 - $b2, 2));
+    }
+
+    public function calculateSimilarity($color1, $color2): float
```

Comment:
@D4lv1k let's add type-hints for function parameters. Also, shouldn't this function have `private` visibility?